src/java/org/apache/nutch/segment/SegmentReader.java

```diff
@@ -70,32 +69,31 @@
 import org.apache.nutch.util.HadoopFSUtil;
 import org.apache.nutch.util.NutchConfiguration;
 import org.apache.nutch.util.NutchJob;
+import org.apache.nutch.util.SegmentReaderUtil;
 
 /** Dump the content of a segment. */
-public class SegmentReader extends Configured implements Tool,
-    Reducer<Text, NutchWritable, Text, Text> {
+public class SegmentReader extends Configured implements Tool {
 
   private static final Logger LOG = LoggerFactory
       .getLogger(MethodHandles.lookup().lookupClass());
 
-  long recNo = 0L;
+  static long recNo = 0L;
 
   private boolean co, fe, ge, pa, pd, pt;
 
-  public static class InputCompatMapper extends MapReduceBase implements
+  public static class InputCompatMapper extends
       Mapper<WritableComparable<?>, Writable, Text, NutchWritable> {
-    private Text newKey = new Text();
+    private static Text newKey = new Text();
```

Comment:
This will cause trouble when the mapper class is used by multiple threads (see MultithreadedMapper).